atecc608a_se.c

```diff
@@ -403,35 +386,44 @@ psa_status_t atecc608a_read(uint16_t slot, size_t offset, uint8_t *data, size_t
     return status;
 }
 
-#define PSA_ATECC608A_LIFETIME 0xdeadbeefU
-
-static psa_drv_se_asymmetric_t atecc608a_asymmetric =
+psa_status_t atecc608a_check_slot(psa_drv_se_context_t *drv_context,
+                                  const psa_key_attributes_t *attributes,
+                                  psa_key_slot_number_t key_slot)
 {
+    if (PSA_KEY_TYPE_IS_ECC_KEY_PAIR(attributes->type)) {
+        if (key_slot <= 15) {
+            return PSA_SUCCESS;
+        }
+    } else if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(attributes->type)) {
+        if (key_slot >= 8 && key_slot <= 15) {
+            return PSA_SUCCESS;
+        }
+    }
+    return PSA_ERROR_NOT_SUPPORTED;
+}
+
+static psa_drv_se_asymmetric_t atecc608a_asymmetric = {
     .p_sign = atecc608a_asymmetric_sign,
     .p_verify = atecc608a_asymmetric_verify,
     .p_encrypt = 0,
     .p_decrypt = 0,
 };
 
-static psa_drv_se_key_management_t atecc608a_key_management =
-{
+static psa_drv_se_key_management_t atecc608a_key_management = {
     /* So far there is no public key import function in the API, so use this instead */
     .p_import = atecc608a_import_public_key,
     .p_generate = atecc608a_generate_key,
     .p_destroy = 0,
-    /* So far there is no public key export function in the API, so use this instead */
-    .p_export = atecc608a_export_public_key,
+    .p_export_public = atecc608a_export_public_key,
+    .p_check_slot = atecc608a_check_slot,
 };
 
-psa_drv_se_info_t atecc608a_drv_info =
-{
-    .lifetime = PSA_ATECC608A_LIFETIME,
-    .p_key_management = &atecc608a_key_management,
-    .p_mac = 0,
-    .p_cipher = 0,
-    .p_asym = &atecc608a_asymmetric,
-    .p_aead = 0,
-    .p_derive = 0,
-    .slot_min = 0,
-    .slot_max = 0,
+psa_drv_se_t atecc608a_drv_info = {
+    .key_management = &atecc608a_key_management,
+    .mac = 0,
```

Comment:
Cosmetic: `NULL` may be more clear here instead of the zeroes.

Reply:
If you want to match types: `NULL` is for pointers to data and `0` is for function pointers. So, `NULL` is good here.